src/constants/logURLTemplates.ts

```diff
@@ -101,40 +103,26 @@ const getResmokeLogURL = (
   return `${logkeeperURL}/build/${buildID}/all?${stringifyQuery(params)}`;
 };
 
-// Although this does not follow typical enum capitalization conventions, it is used to map the origin of the log to the type parameter in the Evergreen URL.
-// Any changes to these origins should be reflected in an ADR
-enum OriginToType {
-  agent = "E",
-  system = "S",
-  task = "T",
-  all = "ALL",
+export enum Origin {
+  Agent = "agent",
+  System = "system",
+  Task = "task",
+  All = "all",
 }
 
-/**
- *
- * @param taskID - the task ID
- * @param execution - the execution number of the task
- * @param origin - the origin of the log
- * @param options - the options for the task log
- * @param options.priority - returned log includes a priority prefix on each line
- * @param options.text - returns the raw log associated with the task
- * @returns an Evergreen URL of the format `/task/${taskID}/${execution}?type=${OriginToType[origin]}&text=true`
- */
 const getEvergreenTaskLogURL = (
-  taskID: string,
-  execution: string | number,
-  origin: keyof typeof OriginToType,
-  options: { priority?: boolean; text?: boolean }
+  logLinks: TaskType["logs"],
+  origin: string,
+  params: { [key: string]: any } = {}
```

Comment:
does it make sense to type the params a bit more strongly?

Reply:
This seemed to be the general typing we use for query params, à la `useQueryParams` https://github.com/evergreen-ci/parsley/blob/4126a3928d37da614d211fb0ecb3ef2735ac0ab6/src/hooks/useQueryParam/index.ts#L17